packages/gatsby-theme-carbon/src/components/ResourceCard/ResourceCard.js

```diff
@@ -1,112 +1,184 @@
-            {cardContainer}
+  // This represents the variant of Resource Card: default, disabled, dark, and mini.
+  const ResourceCardClassNames = classnames(`${prefix}--resource-card`, {
+    [className]: className,
+    [`${prefix}--resource-card--disabled`]: disabled,
+    [`${prefix}--resource-card--dark`]: color === 'dark',
+    [`${prefix}--resource-card--mini`]: type === 'mini',
+  });
+
+  // This gives the Resource Card it's properties for handling links
+  const carbonTileclassNames = classnames(`${prefix}--tile`, {
+    [`${prefix}--tile--clickable`]: href !== undefined,
+  });
+
+  // These are all the icons available to the resource card.
+  const icons = (
+    <>
+      {actionIcon === 'launch' && !disabled ? (
+        <Launch20 aria-label="Open resource" />
+      ) : null}
+      {actionIcon === 'arrowRight' && !disabled ? (
+        <ArrowRight20 aria-label="Open resource" />
+      ) : null}
+      {actionIcon === 'download' && !disabled ? (
+        <Download20 aria-label="Download" />
+      ) : null}
+      {actionIcon === 'email' && !disabled ? (
+        <Email20 aria-label="Email" />
+      ) : null}
+      {actionIcon === 'calendar' && !disabled ? (
+        <Calendar20 aria-label="Calendar" />
+      ) : null}
+      {actionIcon === 'disabled' || disabled === true ? (
+        <Error20 aria-label="disabled" />
+      ) : null}
+    </>
+  );
+
+  // This determines the size of the title prop. productive-heading-03 for default and body-long-02 for Mini.
+  const titleSize = (
+    <>
+      {type === 'default'
+        ? title && (
+            <h4 className={`${prefix}--resource-card__title`}>{title}</h4>
+          )
+        : null}
+      {type === 'mini'
+        ? title && (
+            <h4 className={`${prefix}--resource-card__mini-title`}>{title}</h4>
+          )
+        : null}
+    </>
+  );
+
+  // This determines how and where the icons and image (children) are placed on the page.
+  const iconsAndImages = (
+    <>
+      {type === 'default'
+        ? icons &&
+          children && (
+            <>
+              <div className={`${prefix}--resource-card__icon--img`}>
+                {children}
+              </div>
+              <div className={`${prefix}--resource-card__icon--action`}>
+                {icons}
+              </div>
+            </>
+          )
+        : null}
+      {type === 'mini' ? (
+        children === undefined ? (
+          <div className={`${prefix}--resource-card__mini-icon--action`}>
+            {icons}
           </div>
-        </div>
-      </div>
+        ) : (
+          <div className={`${prefix}--resource-card__mini-icon--img`}>
+            {children}
+          </div>
+        )
+      ) : null}
+    </>
+  );
+
+  // This holds the subtitle, title, icons and images.
+  const cardContent = (
+    <>
+      {subTitle && (
+        <h5 className={`${prefix}--resource-card__subtitle`}>{subTitle}</h5>
+      )}
+      {titleSize}
```

Comment:
Related to my above comment
```suggestion
      {title && (<h4 className={titleClassNames}> {title} </h4> )}
```